### PR TITLE
LFO: document some things that trip up users

### PR DESCRIPTION
**Pull request by jepler**
Closes: #9073

Closes: #9063

ping @kevinjwalters

### CODE REVIEW

File: shared-bindings/synthio/LFO.c [hunk 1]
```diff
@@ -57,7 +57,30 @@ STATIC const uint16_t triangle[] = {0, 32767, 0, -32767};
 //|
 //|     In the current implementation, LFOs are updated every 256 samples. This
 //|     should be considered an implementation detail, though it affects how LFOs
-//|     behave for instance when used to implement an integrator (``l.offset = l``).
+//|     behave for instance when used to implement an integrator (``l.offset = l``).A
```
Comment (dhalbert) [inline shared-bindings/synthio/LFO.c:60]: Stray letter.
```suggestion
//|     behave for instance when used to implement an integrator (``l.offset = l``).
```

File: shared-bindings/synthio/LFO.c [hunk 2]
```diff
@@ -57,7 +57,30 @@ STATIC const uint16_t triangle[] = {0, 32767, 0, -32767};
...
+//|
+//|     An LFO's output is not updated in any other way than when its associated
+//|     synthesizer updates it. For instance, if an LFO is created and its "first"
+//|     output is non-zero, its output is 0 until it is updated by its associated synthesizer.
```
Comment (dhalbert) [inline shared-bindings/synthio/LFO.c:64]: I'm confused. did you mean if its first output is zero?

**Review by dhalbert — CHANGES_REQUESTED** [review]
Will also await @kevinjwalters' comments.

File: shared-bindings/synthio/LFO.c (hunk 2 above)
Comment (jepler) [reply shared-bindings/synthio/LFO.c:64]: Yeah, this is unclear.

A reasonable expectation is that for offset=0, gain=1, the LFO would start with its "value" property equal to waveform[0]. But it doesn't, the its .value property is 0 until it is updated by actually generating samples for some synthesizer.

Comment (jepler) [reply shared-bindings/synthio/LFO.c:64]: I'll re-word this but probably not tonight.

**Review by dhalbert — APPROVED** [review]
This is clear to me now, thanks!

**Comment by jepler** [comment]
changed sawtooth -> triangle in description of LFO once=False

**Review by dhalbert — APPROVED** [review]

**Comment by dhalbert** [comment]
@kevinjwalters How is this for you? Will merge if you find it fixes the explication.

**Comment by kevinjwalters** [comment]
I've been away. Yes, just read through and it's clear, I like the use of an illustrated example.

I'd say the LFO in its current state isn't usable for certain cases. If this was analogue electronics it might be glitchy in an interesting way but it creates harsh transitions from value spurious value of `0`.

**Pull request merged**